modules/@apostrophecms/image/ui/apos/components/AposMediaManagerDisplay.vue

```diff
@@ -95,6 +105,72 @@ export default {
         this.$emit('change', val);
       }
     }
+  },
+  mounted() {
+    // Get the acceptable file types, if set.
+    const imageGroup = apos.modules['@apostrophecms/attachment'].fileGroups
+      .find(group => group.name === 'images');
+
+    if (imageGroup && this.$refs['apos-upload-input']) {
+      const acceptTypes = imageGroup.extensions.map(type => `.${type}`)
+        .join(',');
+      console.info(acceptTypes);
+      this.$refs['apos-upload-input'].setAttribute('accept', acceptTypes);
+    }
+  },
+  methods: {
+    async uploadMedia (event) {
+      const file = event.target.files[0];
+
+      const emptyDoc = await apos.http.post(this.moduleOptions.action, {
+        body: {
+          _newInstance: true
+        }
+      });
+
+      // While the upload is working, set an uploading animation.
+      await this.insertImage(file, emptyDoc);
+
+      // When complete, refresh the image grid, with the new images at top.
+      this.$emit('uploaded');
+
+      // If uploading one image, when complete, load up the edit schema in the right rail.
+      // TODO: Else if uploading multiple images, show them as a set of selected images for editing.
+    },
+    async insertImage(file, emptyDoc) {
+      const formData = new window.FormData();
+
+      formData.append('file', file);
+      let attachment;
+
+      // Make an async request to upload the image.
+      try {
+        attachment = await apos.http.post('/api/v1/@apostrophecms/attachment/upload', {
+          busy: true,
+          body: formData
+        });
+      } catch (error) {
+        console.error('Error uploading media.', error);
+        // apos.notify('Error uploading media.');
+        return;
+      }
+
+      const imageData = Object.assign(emptyDoc, {
+        title: attachment.title,
+        attachment
+      });
+
+      try {
+        await apos.http.post(this.moduleOptions.action, {
```

Comment:
Doesn't look like you did that "open in right rail" thing, do we have a ticket for it to come next?

Reply:
Also a TODO